compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -91,6 +91,18 @@ impl From<P<Expr>> for LhsExpr {
     }
 }
 
+#[derive(Debug)]
+enum DestructuredFloat {
+    /// 1e2
+    Single(Symbol, Span),
+    /// 1.
+    TrailingDot(Symbol, Span, Span),
+    /// 1.2 | 1.2e3
+    WithDot(Symbol, Span, Span, Symbol, Span),
```

Comment:
```suggestion
    MiddleDot(Symbol, Span, Span, Symbol, Span),
```
?

`TrailingDot` also has a dot.